File: Java/Zlozonosc czasowa.java
import java.util.Scanner;

public class Main 
{
    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);
        int k = input.nextInt();
        int maxDepth = 0;
        int currentDepth = 0;
        for (int i = 0; i < k; i++) 
        {
            String line = input.next();
            if (line.equals("for")) 
            {
                currentDepth++;
                maxDepth = Math.max(maxDepth, currentDepth);
            } 
            else if (line.equals("end"))
                currentDepth--;
        }
        if (maxDepth == 0)
            System.out.print("O(1)");
        else if (maxDepth == 1)
            System.out.print("O(n)");
        else
            System.out.print("O(n^" + maxDepth + ")");
        input.close();
    }
}

File: Python/Zlozonosc czasowa.py
k = int(input())
max_depth = 0
current_depth = 0
for _ in range(k):
    line = input()
    if line == 'for':
        current_depth += 1
        max_depth = max(max_depth, current_depth)
    elif line == 'end':
        current_depth -= 1
if max_depth == 0:
    print('O(1)')
elif max_depth == 1:
    print('O(n)')
else:
    print(f'O(n^{max_depth})')